azurelinuxagent/common/rdma.py

```diff
@@ -265,11 +265,144 @@ def provision_network_direct_rdma(self):
         RDMADeviceHandler.update_network_interface(self.mac_addr, self.ipv4_addr)
 
     def provision_sriov_rdma(self):  # pylint: disable=R1711
-        RDMADeviceHandler.wait_any_rdma_device(
-            self.sriov_dir, self.device_check_timeout_sec, self.device_check_interval_sec)
-        RDMADeviceHandler.update_iboip_interface(self.ipv4_addr, self.ipoib_check_timeout_sec,
-                                                 self.ipoib_check_interval_sec)
-        return
+
+        (key, value) = self.read_ipoib_data()
+        if key:
+            # provision multiple IP over IB addresses
+            logger.info("RDMA: provisioning multiple IP over IB addresses")
+            self.provision_sriov_multiple_ib(value)
+        elif self.ipv4_addr:
+            logger.info("RDMA: provisioning single IP over IB address")
+            # provision a single IP over IB address
+            RDMADeviceHandler.wait_any_rdma_device(self.sriov_dir,
+                self.device_check_timeout_sec, self.device_check_interval_sec)
+            RDMADeviceHandler.update_iboip_interface(self.ipv4_addr,
+                self.ipoib_check_timeout_sec, self.ipoib_check_interval_sec)
+        else:
+            logger.info("RDMA: missing IP address")
+
+    def read_ipoib_data(self) :
+
+        # read from KVP pool 0 to figure out the IP over IB addresses
+        kvp_key_size = 512
+        kvp_value_size = 2048
+        driver_info_source = '/var/lib/hyperv/.kvp_pool_0'
+
+        if not os.path.isfile(driver_info_source):
+            logger.error("RDMA: can't read KVP pool 0")
+            return (None, None)
+
+        key_0 = None
+        value_0 = None
+        f = open(driver_info_source, "rb")  # pylint: disable=C0103
```

Comment:
Please take care of the pylint warning (#C0103) [rename `f` to a more meaningful name like `driver_file` or something)]